src/egraph.rs

```diff
@@ -202,6 +202,58 @@ impl<L: Language> EGraph<L> {
         pruned
     }
 
+    pub fn fold_constants(&mut self) -> usize {
+        let mut folded = 0;
+        let mut to_add = HashMap::default();
+        // keep track of constants in each class to save some scanning
+        let mut constant_nodes = HashMap::default();
+
+        // look for constants in each class
+        for (id, class) in &self.classes {
+            for node in &class.nodes {
+                if let Expr::Constant(c) = node {
+                    constant_nodes.insert(*id, c.clone());
```

Comment:
This still raises the question of what happens if there is more than one constant in an eclass.

Maybe for now the right thing to do is to just assert that there's only one.